tests/PySys/monitoring/monitoring_smaller_interval/run.py

```diff
@@ -0,0 +1,137 @@
+            stdouterr="tedge_sub",
+            background=True,
+        )
+
+        # Wait for a small amount of time to give tedge sub time
+        # to initialize. This is a heuristic measure.
+        # Without an additional wait we observe failures in 1% of the test
+        # runs.
+        time.sleep(0.1)
+
+        for i in range(10):
+
+            pub = self.startProcess(
+                command=self.sudo,
+                arguments=[self.tedge, "mqtt", "pub",
+                           "collectd/host/temperature/temp", "123435445:25.5"],
+                stdouterr="tedge_temp",
+            )
+
+            pub = self.startProcess(
+                command=self.sudo,
+                arguments=[self.tedge, "mqtt", "pub",
+                           "collectd/host/pressure/pres", "12345678:500.5"],
+                stdouterr="tedge_pres",
+            )
+
+            # publish every 100ms
+            time.sleep(0.1)
+
+        # wait for tedge-dm-agent to batch messages
+        time.sleep(1)
+
+        # Kill the subscriber process explicitly with sudo as PySys does
+        # not have the rights to do it
+        kill = self.startProcess(
+            command=self.sudo,
+            arguments=["killall", "tedge"],
+            stdouterr="kill_out",
+        )
+
+    def validate(self):
+        self.assertThat('collectd_msg_validation_result == expected_result',
+                        collectd_msg_validation_result=self.validate_json(), expected_result=True)
+
+    def validate_json(self):
+        f = open(self.output + '/tedge_sub.out', 'r')
+        lines = f.readlines()
+        for line in lines:
+            self.js_msg = json.loads(line)
+            if not self.validate_time():
+                reason = "time validation failed in message: " + str(line)
+                self.abort(False, reason)
+            if not self.validate_temperature():
+                reason = "temperature stat validation failed in message: " + \
+                    str(line)
+                self.abort(False, reason)
+            if not self.validate_pressure():
+                reason = "pressure stat validation failed in message: " + \
+                    str(line)
+                self.abort(False, reason)
+        return True
+
+    def validate_time(self):
+        if self.js_msg["time"]:
+            return True
+        else:
+            return False
+
+    def validate_temperature(self):
+        if self.js_msg["temperature"]:
+            if "temp" in self.js_msg["temperature"]:
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    def validate_pressure(self):
+        if self.js_msg["pressure"]:
+            if "pres" in self.js_msg["pressure"]:
+                return True
+            else:
+                return False
+        else:
+            return False
+
+    def cleanup(self):
```

Comment:
Naming it `cleanup` will override the `cleanup` function in the `BaseTest`. Name it something else, otherwise the `addCleanup` function logic won't work.